steamclock/steamclog-android
# Added support for setting user id on Sentry reports (#63)

### Related Issue: #63 


### Summary of Problem:
When switching over to Sentry I commented out the ability to associate a userId with subsequent error reports; this PR adds that support back in for Sentry.

### Proposed Solution:
Followed steps here https://docs.sentry.io/platforms/android/enriching-events/identify-user/ to set the proper user object on Sentry; also added function to clear it.

### Testing Steps:
* Run sample app
* Hit the "Set UserId" button, then the "Send Non-Fatal" button
* Restart app to force reports to send
* Check the Sentry console (https://sentry.io/organizations/steamclock-software/issues/2397673130/?project=5399932) and make sure the reports show user id "1234"

![Screen Shot 2021-05-18 at 12 51 42 PM](https://user-images.githubusercontent.com/5217641/118715496-865d1c00-b7d8-11eb-8182-86c408409a6e.png)
